Extensions/ArtifactEngine/E2ETests/E2E.ts

```diff
@@ -47,11 +50,76 @@ describe('e2e tests', () => {
                     assert.equal(data, "dummyFolderContent");
                     done();
                 });
-                
+
                 assert.equal(tickets.find(x => x.artifactItem.path == "Extensions/ArtifactEngine/TestData/Jenkins/file1.pdb").retryCount, 0);
                 assert.equal(tickets.find(x => x.artifactItem.path == "Extensions/ArtifactEngine/TestData/Jenkins/folder1/file2.txt").retryCount, 0);
             }, (error) => {
                 throw "test failure";
             });
     });
+
+    it('should be able to download build artifact', function (done) {
+        this.timeout(15000);
+        let processor = new engine.ArtifactEngine();
+
+        let processorOptions = new engine.ArtifactEngineOptions();
+        processorOptions.fileProcessingTimeoutInMinutes = 5;
+        processorOptions.itemPattern = "**";
+        processorOptions.parallelProcessingLimit = 8;
+        processorOptions.retryIntervalInSeconds = 1000;
+        processorOptions.retryLimit = 2;
+        processorOptions.verbose = true;
+
+        var itemsUrl = "https://testking123.visualstudio.com/_apis/resources/Containers/1898832?itemPath=Dropz&isShallow=false"
+        var variables = {};
+
+        var handler = new PersonalAccessTokenCredentialHandler(config.vsts.pat);
+        var webProvider = new providers.WebProvider(itemsUrl, "vsts.handlebars", variables, handler, { ignoreSslError: false });
+        var dropLocation = path.join(config.dropLocation, "vstsDropWithMultipleFiles");
+        var filesystemProvider = new providers.FilesystemProvider(dropLocation);
+
+        processor.processItems(webProvider, filesystemProvider, processorOptions)
+            .then((tickets) => {
+                fs.readFile(path.join(config.dropLocation, 'vstsDropWithMultipleFiles/dropz/folder1/file2.txt'), 'utf8', function (err, data) {
+                    if (err) {
+                        throw err;
+                    }
+                    assert.equal(data, "dummyFolderContent");
+                    done();
+                });
+
+                assert.equal(tickets.find(x => x.artifactItem.path == "dropz/file1.pdb").retryCount, 0);
+                assert.equal(tickets.find(x => x.artifactItem.path == "dropz/folder1/file2.txt").retryCount, 0);
+                done();
+            }, (error) => {
+                throw "test failure";
+            });
+    });
+
+    it('should be able to download large build artifact', function (done) {
```

Comment:
good one! what is the size of the build artifact?

Reply:
its around 1.7 GBs with 301 files